source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -39,6 +39,15 @@
 from jormungandr import app
 import gevent
 import gevent.pool
+import gevent.socket
+import urllib3
+
+# This line replaces the gevent.monkey.patch_socket()
+# the reason why we don't use patch_socket() at the very beginning of jormungandr is
+# that it caused a mysterious strong performance regression for certain requests, thus we patch
+# only at places where asynchronisation is needed
+urllib3.connection.connection.socket = gevent.socket
```

Comment:
Doing this in experimental will have some strange side effects when this scenario will be imported for the first time. I prefer monkey patch to be done globally, so we have consistent behavior no matter what call have been made previously.